File: src/main/java/net/liyze/basin/core/commands/ShellCommand.java
package net.liyze.basin.core.commands;

import net.liyze.basin.context.annotation.Component;
import net.liyze.basin.core.Command;
import org.jetbrains.annotations.NotNull;

import java.io.IOException;
import java.util.List;

import static net.liyze.basin.core.Main.*;

@Component
public class ShellCommand implements Command {
    @Override
    public void run(@NotNull List<String> args) {
        if (!cfg.enableShellCommand) return;
        try {
            LOGGER.info(Runtime.getRuntime().exec(args.toArray(new String[0])).toString());
        } catch (IOException e) {
            LOGGER.error(e.toString());
        }
    }

    @Override
    public @NotNull String Name() {
        return "exec";
    }
}

File: src/main/kotlin/net/liyze/basin/core/scan/ShellCommand.kt
/*
 * Copyright (c) 2023 Liyze09
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.liyze.basin.core.scan

import net.liyze.basin.Article
import net.liyze.basin.core.Command
import net.liyze.basin.core.LOGGER
import net.liyze.basin.core.cfg
import org.slf4j.Logger
import java.io.IOException


class ShellCommand : Command {
    override fun run(args: List<String?>, logger: Logger, context: Article) {
        if (!cfg.enableShellCommand) {
            logger.warn("Shell execution is disabled!")
            return
        }
        try {
            LOGGER.info(Runtime.getRuntime().exec(args.toTypedArray<String?>()).toString())
        } catch (e: IOException) {
            LOGGER.error(e.toString())
        }
    }

    override fun getName(): String {
        return "exec"
    }
}
